packages/typespec-client-generator-core/src/decorators.ts

```diff
@@ -795,3 +800,18 @@ export function $flattenProperty(context: DecoratorContext, target: ModelPropert
 export function shouldFlattenProperty(context: SdkContext, target: ModelProperty): boolean {
   return getScopedDecoratorData(context, flattenPropertyKey, target) ?? false;
 }
+
+const clientNameKey = createStateSymbol("clientName");
+
+export function $clientName(
+  context: DecoratorContext,
+  entity: Type,
+  value: string,
+  scope?: string
```

Comment:
I forgot if we can use string literal unions for TSP decorator parameters. I was thinking it may be useful to provide hints about possible values.

```ts
export function $clientName(
  context: DecoratorContext,
  entity: Type,
  value: string,
  scope?: "dotnet" | "java" | "python" | "javascript" | "go" | (string & {}))
```

That, in typescript will allow you to input any string but intellisense will suggest the strings in the union. Not sure if this would work with the TSP language server

Reply:
I'm not sure it would, but might as well try

Reply:
tests still pass, not sure how much it will help the UI, but I think it's good to have even just for us. Thanks!